### DISCUSSION TITLE
GraphNode improvements

### DISCUSSION THREAD

**Discussion by ningxia2022** [godotengine/godot-proposals#9902] [GUI]
I strongly recommend expanding the functionality of GraphNode plugins and not abandoning such valuable tools. I would like to suggest adding or enhancing the following features:
1、Enable the slot to support multiple directions: up input, down input, left input, right input, and all corresponding outputs.
It would be best if the slot can support four directions: up input, down, left, right, all outputs, or left input, up, down, right output, or other. Strongly recommend adding this feature
 This flexibility in slot directions would greatly enhance usability.
2、Simplify the process of setting the number of slots. Currently, there is no option to specify the number of slots on the panel easily.
3、I want to reiterate the importance of not giving up on the graphnode node. It has great potential for improvement, so it should not be overlooked.
Please consider these suggestions for improving the plugin features.

**Comment by AThousandShips** [comment]
Converting to a discussion as this was not filled in properly
